packages/loader/container-utils/src/error.ts

```diff
@@ -157,28 +161,52 @@ export function CreateContainerError(originalError: any, props?: ITelemetryPrope
 }
 
 /**
- * Take an unknown error object and extract certain known properties to be included in a new error object.
- * The stack is preserved, along with any safe-to-log telemetry props.
- * @param error - An error that was presumably caught, thrown from unknown origins
+ * Create a new error, wrapping and caused by the given unknown error.
+ * Copies the inner error's message and stack over but otherwise uses newErrorFn to define the error.
+ * The inner error's instance id will also be logged for telemetry analysis.
+ * @param innerError - An error from untrusted/unknown origins
  * @param newErrorFn - callback that will create a new error given the original error's message
  * @returns A new error object "wrapping" the given error
  */
-export function wrapError<T extends IWriteableLoggingError>(
-    error: any,
+export function wrapError<T extends IFluidErrorBase>(
+    innerError: unknown,
     newErrorFn: (m: string) => T,
 ): T {
     const {
         message,
         stack,
-    } = extractLogSafeErrorProperties(error, false /* sanitizeStack */);
-    const props = isILoggingError(error) ? error.getTelemetryProperties() : {};
+    } = extractLogSafeErrorProperties(innerError, false /* sanitizeStack */);
 
     const newError = newErrorFn(message);
-    newError.addTelemetryProperties(props);
 
     if (stack !== undefined) {
-        Object.assign(newError, { stack });
+        // supposedly setting stack on an Error can throw.
+        try {
+            Object.assign(newError, { stack });
```

Comment:
Is there a way we can just not set stack? We should only instantiate Error where they truly occur and define stack as a byproduct.
I buy in to why some clients will throw here. The runtime environment should be authoritative here as to the appropriate value, and allowing it to be set may spread disinformation.

This is now 'sanitizedStack' if a member of the same object instance and is not representative any longer of even the non-standard Mozilla property.

At a glance I believe we should have strived to leave the original error object intact during its lifetime but not log values of its unsafe members directly. Which seems to be everything, except maybe name... Do we log name?

Reply:
Oooh now that we have errorInstanceId we can stop smashing stack here.  Yay.

Reply:
But upon further consideration, I'm going to open an issue for that and leave this as-is.  I think in the spirit of what I want `wrapError` to be, it shouldn't copy the stack.  Maybe not even the message.  I dunno.  It's only used in a few places and some of them are dubious.  So I'm not going to sweat it.

Reply:
Actually I'm not opening as issue.  We'll see how `wrapError` gets used over the coming months and figure it out as we go.